### PR TITLE
Add suport for Administrator Mode

### LINKED ISSUE #10: Administrator mode for helm-chart
I would like to provide cluster wide access for litmus during the HA tests which will be include:
- installation of litmus
- run test
- generate report
- delete litmus.

Could you advice if it's possible to add such option in helm-chart?
I can help with the contribution. Let me know what do you think about that.
thanks!

### PR DESCRIPTION
**Pull request by Jasstkn**
Signed-off-by: Jasstkn <jasssstkn@yahoo.com>

closes #10

### CODE REVIEW

**Comment by Jasstkn** [comment]
@ksatchit could you review that?

**Comment by ksatchit** [comment]
@Jasstkn , I see that we have attempted to change the existing rbac resources (svc account, clusterrole, clusterrolebinding) in case admin mode is enabled. Now, the current/existing rbac and the resources mapped therein are specific to just what the operator needs. Whereas what we would put in the litmus-admin rbac would be a *different* set needed for running the experiments..i.e., to be used in *chaosServiceAccount* field of the chaosengine spec.

While I like the idea of `{{ if .Values.administratorMode.enabled }}`, I feel it should *add* a new set of svaccount/clusterrole/clusterrolebinding and not modify the existing one (and hence by extension of this logic not modify the operator deploy template)

**Comment by Jasstkn** [comment]
> @Jasstkn , I see that we have attempted to change the existing rbac resources (svc account, clusterrole, clusterrolebinding) in case admin mode is enabled. Now, the current/existing rbac and the resources mapped therein are specific to just what the operator needs. Whereas what we would put in the litmus-admin rbac would be a _different_ set needed for running the experiments..i.e., to be used in _chaosServiceAccount_ field of the chaosengine spec.
> 
> While I like the idea of `{{ if .Values.administratorMode.enabled }}`, I feel it should _add_ a new set of svaccount/clusterrole/clusterrolebinding and not modify the existing ones/conditionally add different resources in the role (and hence by extension of this logic not modify the operator deploy template)

I do not quite understand your point. As it was declared in the documentation, we need to create an additional RBAC for admin mode and refer to it in the deployment manifest. (https://docs.litmuschaos.io/docs/admin-mode/) And I did exactly that. If administrator mode enabled then manifests will be created. If at the same time, we have an exporter enabled, then the normal RBAC will be created as well. Please, clarify what exactly do you propose.

**Comment by ksatchit** [comment]
@Jasstkn , I am just pasting some proposed templates here with inline comments. Hope this is understandable: 


[clusterRole.yaml.txt](https://github.com/litmuschaos/litmus-helm/files/4567517/clusterRole.yaml.txt)
[clusterRoleBinding.yaml.txt](https://github.com/litmuschaos/litmus-helm/files/4567518/clusterRoleBinding.yaml.txt)
[serviceaccount.yaml.txt](https://github.com/litmuschaos/litmus-helm/files/4567521/serviceaccount.yaml.txt)
[deployment.yaml.txt](https://github.com/litmuschaos/litmus-helm/files/4567519/deployment.yaml.txt)

**Comment by Jasstkn** [comment]
> @Jasstkn , I am just pasting some proposed templates here with inline comments. Hope this is understandable:
> 
> [clusterRole.yaml.txt](https://github.com/litmuschaos/litmus-helm/files/4567517/clusterRole.yaml.txt)
> [clusterRoleBinding.yaml.txt](https://github.com/litmuschaos/litmus-helm/files/4567518/clusterRoleBinding.yaml.txt)
> [serviceaccount.yaml.txt](https://github.com/litmuschaos/litmus-helm/files/4567521/serviceaccount.yaml.txt)
> [deployment.yaml.txt](https://github.com/litmuschaos/litmus-helm/files/4567519/deployment.yaml.txt)

Hello, @ksatchit I backed with updates according to that PR. I've updated what you have asked me. Could you take a look again?

File: charts/litmus/templates/deployment.yaml [hunk 1]
```diff
@@ -18,18 +18,22 @@ spec:
         name: {{ include "litmus.name" . }}
         instance: {{ .Release.Name }}
     spec:
-      serviceAccountName: {{ include "litmus.fullname" . }} 
+      {{- if not .Values.administratorMode.enabled }}
+      serviceAccountName: {{ include "litmus.fullname" . }}
+      {{- else if .Values.administratorMode.enabled }}
+      serviceAccountName: {{ include "litmus.fullname" . }}-admin
+      {{- end }}
```
Comment (ksatchit) [inline charts/litmus/templates/deployment.yaml:25]: This change won't be needed. As explained, the operator will continue to use just litmus. The litmus-admin will be reserved purely for the chaosengine.

**Comment by ksatchit** [comment]
Thanks @Jasstkn ! Looks good, just had one minor comment on the deployment.

File: charts/litmus/templates/deployment.yaml (hunk 1 above)
Comment (Jasstkn) [reply charts/litmus/templates/deployment.yaml:25]: Ahhh, I see. Ok, will change that, thanks!

**Pull request merged**